File: solutions/typescript/darts/2/darts.ts
export function score(x: number, y: number): number {
    const distance = Math.hypot(x, y);
    const scores = new Map([
        [distance <= 10, 1],
        [distance <= 5, 5],
        [distance <= 1, 10],
    ]);
    return scores.get(true) ?? 0;
}

File: solutions/python/darts/2/darts.py
import math


def score(x: int | float, y: int | float):
    distance = math.sqrt(x**2 + y**2)

    in_inner = distance <= 1
    in_middle = distance > 1 and distance <= 5
    in_outer = distance > 5 and distance <= 10

    return (10 * in_inner) + (5 * in_middle) + (1 * in_outer)
